docs/user-guides/federation.md

```diff
@@ -0,0 +1,154 @@
+# Federate Monitoring Stack and Openshift In-Cluster Prometheus
+
+## Architecture / Topology
+
+![Architecture](federation/assets/cmo-obo-federation.png)
+
+This example deploy a MonitoringStack in `federate-cmo` namespace and reads
+only a selected set of metrics from the in-cluster Prometheus.
+
+## Steps
+
+Assuming that ObO is installed, follow the steps below to create a
+MonitoringStack that uses the in-cluster Prometheus `/federate` endpoint to
+scrape selected metrics.
+
+**NOTE:** All examples used in this guide can be found under the [manifests
+directory](federation/manifests)
+
+### Create a Project to deploy MonitoringStack
+
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: federate-cmo
+spec:
+```
+
+
+### Deploy Monitoring Stack
+
+
+Apply the following MonitoringStack
+
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: federate-cmo-ms
+  namespace: federate-cmo
+spec:
+
+  resourceSelector:  # 👈 this is used to select the ServiceMonitor
+    matchLabels:
+      monitoring.rhobs/stack: federate-cmo-ms
+
+  logLevel: info # 👈 debug for verbose logs
+  retention: 3h
+
+  prometheusConfig:
+    replicas: 2  # 👈 ensures that at least one prometheus is running during upgrade
+
+    enableRemoteWriteReceiver: true
+
+  alertmanagerConfig:
+    disabled: true
+
+  resources: # 👈 ensure that you provide sufficient amount of resources
+    requests:
+      cpu: 500m
+      memory: 1Gi
+```
+
+
+### Grant permission to Federate In-Cluster Prometheus
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: federate-cmo-ms-view
+  labels:
+    kubernetes.io/part-of: federate-cmo-ms
+    monitoring.rhobs/stack: federate-cmo-ms
+
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+subjects:
+- kind: ServiceAccount
+  name: federate-cmo-ms-prometheus  # 👈  ServiceAccount used in the prometheus deployed by ObO
+  namespace: federate-cmo
+```
+
+### Create ServiceMonitor for Federation
+
+```yaml
+apiVersion: monitoring.rhobs/v1
+kind: ServiceMonitor
+metadata:
+  name: federate-cmo-smon
+  namespace: federate-cmo
+  labels:
+    kubernetes.io/part-of: federate-cmo-ms
+    monitoring.rhobs/stack: federate-cmo-ms
+
+spec:
+  selector: # 👈 use the prometheus service to create a "dummy" target.
+    matchLabels:
+      app.kubernetes.io/managed-by: observability-operator
+      app.kubernetes.io/name: federate-cmo-ms-prometheus
```

Comment:
Does this point the servicemonitor the the CMO prometheus instance?

Reply:
No, The use of the service is only to create "dummy" target which will be used to scrape off the incluster prometheus.

Reply:
Can you elaborate more on how it works? Does it require any additional configs not included in the document?

Reply:
@yftacherzog the documentation should be complete

Reply:
> Can you elaborate more on how it works? 

We configure a MS that watches a service-monitors (see matchLabels) in the same ns as the MS. 

The ServiceMonitor needs  "some" target - right? 
For that we use the same service as the one created for the MonitoringStack (federate-cmo Service) but then we hijack that servicemonitor to then set the endpoints to `serverName: prometheus-k8s.openshift-monitoring.svc`. We add `relabeling` to replace `__address__` with `prometheus-k8s.openshift-monitoring.svc`

A good way to understand this would be to apply the manifests and modify each setting to see its affect.

Reply:
@simonpasquier please feel free to correct me :)

Reply:
Thanks for the explanation. I'll give it another try early next week.

What's the benefit of this approach compared to your original one?

Reply:
> What's the benefit of this approach compared to your original one?

there's no need to deploy a service monitor (or any resource) in the openshift-monitoring namespace.